src/subplots.ts

```diff
@@ -1358,10 +1477,523 @@ export class PrimitiveGroupContainer extends PlotData {
+        ctrlKey = false;
+      });
+    }
+  }
+
+  public wheel_interaction(mouse3X, mouse3Y, e: WheelEvent) { //TODO: TO REFACTOR !!!
+    // e.preventDefault();
+    this.fusion_coeff = 1.2;
+    var event = -Math.sign(e.deltaY);
+    mouse3X = e.offsetX;
+    mouse3Y = e.offsetY;
+    if ((mouse3Y>=this.height - this.decalage_axis_y + this.Y) && (mouse3X>this.decalage_axis_x + this.X) && this.axis_ON) {
+        if (event>0) {
+          this.scaleX = this.scaleX*this.fusion_coeff;
+          this.scroll_x++;
+          this.originX = this.width/2 + this.fusion_coeff * (this.originX - this.width/2);
+        } else if (event<0) {
+          this.scaleX = this.scaleX/this.fusion_coeff;
+          this.scroll_x--;
+          this.originX = this.width/2 + 1/this.fusion_coeff * (this.originX - this.width/2);
+        }
+
+    } else if ((mouse3X<=this.decalage_axis_x + this.X) && (mouse3Y<this.height - this.decalage_axis_y + this.Y) && this.axis_ON) {
+        if (event>0) {
+          this.scaleY = this.scaleY*this.fusion_coeff;
+          this.scroll_y++;
+          this.originY = this.height/2 + this.fusion_coeff * (this.originY - this.height/2);
+        } else if (event<0) {
+          this.scaleY = this.scaleY/this.fusion_coeff;
+          this.scroll_y--;
+          this.originY = this.height/2 + 1/this.fusion_coeff * (this.originY - this.height/2);
+        }
+
+    } else {
+        if (event>0)  var coeff = this.fusion_coeff; else coeff = 1/this.fusion_coeff;
+        this.scaleX = this.scaleX*coeff;
+        this.scaleY = this.scaleY*coeff;
+        this.scroll_x = this.scroll_x + event;
+        this.scroll_y = this.scroll_y + event;
+        this.originX = mouse3X - this.X + coeff * (this.originX - mouse3X + this.X);
+        this.originY = mouse3Y - this.Y + coeff * (this.originY - mouse3Y + this.Y);
+      }
+      if (isNaN(this.scroll_x)) this.scroll_x = 0;
+      if (isNaN(this.scroll_y)) this.scroll_y = 0;
+      return [mouse3X, mouse3Y];
+  }
+}
+
+export class Frame extends BasePlot {
+  public xFeature: string;
+  public yFeature: string;
+  protected _nXTicks: number;
+  protected _nYTicks: number;
+  readonly OFFSET = new Vertex(80, 50);
+  readonly MARGIN = new Vertex(20, 20);
+  constructor(
+    public data: any,
+    public width: number,
+    public height: number,
+    public buttons_ON: boolean,
+    public X: number,
+    public Y: number,
+    public canvas_id: string,
+    public is_in_multiplot: boolean = false
+    ) {
+      super(data, width, height, buttons_ON, X, Y, canvas_id, is_in_multiplot);
+      [this.xFeature, this.yFeature] = this.setFeatures(data);
+      this.axes = this.setAxes();
+      this.fixedObjects.push(...this.axes);
+    }
+
+  get movingMatrix(): DOMMatrix {
+    const movingMatrix = this.axes[0].transformMatrix;
+    movingMatrix.d = this.axes[1].transformMatrix.d;
+    return this.canvasMatrix.multiply(movingMatrix)
+  }
+
+  get nXTicks() {return this._nXTicks ? this._nXTicks : 7}
```

Comment:
`_nYTicks` and `_nXTicks` could define their default value to 7 and this would allow us to get rid of getters and setters

Reply:
I think i missed something with inheritance stuff. Here, if i set them with a default value, the inherited Histogram wnt be able to overwrite them with its own default values.

Reply:
If they are private in the base class, they should need to be "protected" instead. Otherwise, i missed something as well